Turn off MPU for PSOC 6

c1728p9 opened the pull request:
### Description

Turn off  the MPU for the PSOC 6 since it has a non-standard memory map. This fixes crashes on boot for this target.

fixes #8937

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

c1728p9 committed:
Turn off MPU for PSOC 6

Turn off  the MPU for the PSOC 6 since it has a non-standard memory
map. This fixes crashes on boot for this target.

alzix commented on targets/targets.json:
```diff
@@ -7582,25 +7583,21 @@
         "inherits": ["MCU_PSOC6"],
         "core": "Cortex-M0+",
-        "macros": ["MCU_PSOC6_M0"],
+        "macros_add": ["MCU_PSOC6_M0"],
```
i think this one is not related to our MPU configuration, but clearly a necessary change!
Can you consider splitting it to separate commit?

c1728p9 replied:
done

alzix reviewed (approved):
LGTM

alzix commented:
fixes #8937 - once merged will close the issue automatically

c1728p9 committed:
Add rather than override PSOC 6 macros

Use "macros_add" rather than "marcos" so any macros defined by
MCU_PSOC6 are not overwritten.

cmonr commented:
CI started
